### PR TITLE
Add v22.2.4 release notes

### PR DESCRIPTION
**Pull request by rmloveland**
Fixes DOC-4613

### CODE REVIEW

File: _includes/releases/v22.2/v22.2.4.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+- Fixed a bug where a DNS lookup was performed during gossip remote forwarding while holding the gossip mutex. This could cause processing stalls if the DNS server was slow to respond, since we need to acquire gossip read locks in several performance critical code paths, including [Raft processing](../v22.2/architecture/replication-layer.html#raft). [#95443][#95443]
+- Operations like [`BACKUP`](../v22.2/backup.html) can now reuse a previously created [AWS KMS client](../v22.2/take-and-restore-encrypted-backups.html#use-key-management-service) if the client was created with the same parameters. This addresses the `NoCredentialProviders` errors on EC2 with for backups with long incremental chains. [#95534][#95534]
+- Previously, [trigrams](../v22.2/trigram-indexes.html) ignored Unicode (multi-byte) characters from input strings. This is now corrected. [#94199][#94199]
```
Comment (ianjevans) [inline _includes/releases/v22.2/v22.2.4.md:49]:
```suggestion
- Fixed a bug where [trigrams](../v22.2/trigram-indexes.html) ignored Unicode (multi-byte) characters from input strings. [#94199][#94199]
```

File: _includes/releases/v22.2/v22.2.4.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+- Fixed the `array_to_string` [built-in function](../v22.2/functions-and-operators.html#array_to_string) so that nested arrays are traversed without printing 'ARRAY' at each nesting level. [#95843][#95843]
+- Fixed a bug in which [`RESTORE SYSTEM USERS`](../v22.2/restore.html) would fail to restore [role](../v22.2/security-reference/authorization.html#users-and-roles) options. [#95293][#95293]
+- Fixed a bug causing [ranges](../v22.2/architecture/overview.html#architecture-range) to remain without a [leaseholder](../v22.2/architecture/replication-layer.html#leases) in cases of asymmetric [network partitions](../v22.2/cluster-setup-troubleshooting.html#network-partition). [#95221][#95221]
+- Fixed a bug whereby a [stalled disk](../v22.2/cluster-setup-troubleshooting.html#disk-stalls) would sometimes be undetected, hanging the Cockroach process indefinitely. Now the stall is detected and the process is terminated if the `storage.max_sync_duration.fatal.enabled` [cluster setting](../v22.2/cluster-settings.html) is enabled, and the stall is observed to last longer than the value in the `storage.max_sync_duration` cluster setting. [#96036][#96036]
```
Comment (ianjevans) [inline _includes/releases/v22.2/v22.2.4.md:53]:
```suggestion
- Fixed a bug whereby a [stalled disk](../v22.2/cluster-setup-troubleshooting.html#disk-stalls) would sometimes be undetected, hanging the CockroachDB process indefinitely. Now the stall is detected and the process is terminated if the `storage.max_sync_duration.fatal.enabled` [cluster setting](../v22.2/cluster-settings.html) is enabled, and the stall is observed to last longer than the value in the `storage.max_sync_duration` cluster setting. [#96036][#96036]
```

**Review by ianjevans — APPROVED** [review]
Just some nits on wording as suggestions. Take them or leave them.

File: _includes/releases/v22.2/v22.2.4.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+- The `confluent_schema_registry` URI for [avro changefeeds](../v22.2/changefeed-messages.html#avro) now supports the `client_cert` and `client_key` params. [#96533][#96533]
+
+<h3 id="v22-2-4-sql-language-changes">SQL language changes</h3>
+
+- The [`SHOW GRANTS ON EXTERNAL CONNECTION`](../v22.2/show-grants.html) and [`SHOW SYSTEM GRANTS`](../v22.2/show-system-grants.html) commands now use a column name of `privilege_type` rather than `privilege`. For external connections, the `name` column was changed to `connection_name`. This makes the commands consistent with other `SHOW GRANTS` commands. [#95567][#95567]
+- If `copy_from_retries_enabled` is set, [`COPY`](../v22.2/copy-from.html) is now able to retry under certain safe circumstances - namely when `copy_from_atomic_enabled` is false, there is no [transaction](../v22.2/transactions.html) running `COPY`, and the error returned is retryable. This prevents users who keep running into [`TransactionProtoWithRefreshError`](../v22.2/transaction-retry-error-reference.html) from having issues. [#95496][#95496]
+- Improve the performance of [trigram operations](../v22.2/trigram-indexes.html). [#94199][#94199]
```
[the remark this answers is not in the document]
Comment (rmloveland) [reply _includes/releases/v22.2/v22.2.4.md:21]: Fixed thanks!

File: _includes/releases/v22.2/v22.2.4.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+{% include releases/release-downloads-docker-image.md release=include.release %}
+
+<h3 id="v22-2-4-security-updates">Security updates</h3>
+
+- Added the `COCKROACH_TLS_ENABLE_OLD_CIPHER_SUITES` environment variable, which re-enables some less secure TLS 1.2 cipher suites for use with legacy clients. [#95705][#95705]
+
+<h3 id="v22-2-4-{{-site.data.products.enterprise-}}-edition-changes">{{ site.data.products.enterprise }} edition changes</h3>
+
+- Fixed a bug in [`ALTER CHANGEFEED`](../v22.2/alter-changefeed.html) that would panic when altering a changefeed to remove a table that had already been dropped. [#95785][#95785]
+- Update memory accounting for [changefeeds](../v22.2/create-and-configure-changefeeds.html) to improve cluster stability. [#96182][#96182]
```
[the remark this answers is not in the document]
Comment (rmloveland) [reply _includes/releases/v22.2/v22.2.4.md:14]: Fixed!

File: _includes/releases/v22.2/v22.2.4.md [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+- Improve the performance of [trigram operations](../v22.2/trigram-indexes.html). [#94199][#94199]
+- [`COPY ... FROM ... QUOTE '"'`](../v22.2/copy-from.html) will no longer error. [#96766][#96766]
+
+<h3 id="v22-2-4-operational-changes">Operational changes</h3>
+
+- Histogram metrics in the [DB Console](../v22.2/ui-overview.html) (such as on the [SQL Dashboard](../v22.2/ui-sql-dashboard.html), for example) can now optionally use the legacy `HdrHistogram` model by setting the environment variable `COCKROACH_ENABLE_HDR_HISTOGRAMS=true` on CockroachDB nodes. **Note that this is not recommended** unless users are having difficulties with the newer Prometheus-backed histogram model. Enabling can cause performance issues with timeseries databases like Prometheus, as processing and storing the increased number of buckets is taxing on both CPU and storage. Note that the `HdrHistogram` model is slated for full deprecation in an upcoming release. [#96514][#96514]
```
[the remark this answers is not in the document]
Comment (rmloveland) [reply _includes/releases/v22.2/v22.2.4.md:26]: Fixed!

File: _includes/releases/v22.2/v22.2.4.md [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+- Prometheus histograms will now export more buckets across the board to improve precision & fidelity of information reported by histogram metrics, such as quantiles, as shown in multiple pages in the [DB Console](../v22.2/ui-overview.html). This will lead to an increase in storage requirements to process these histogram metrics in downstream systems like Prometheus, but should still be a marked improvement when compared to the legacy `HdrHistogram` model. If users have issues with the precision of these bucket boundaries, they can set the environment variable `COCKROACH_ENABLE_HDR_HISTOGRAMS=true` to revert to using the legacy `HdrHistogram` model instead, although this is not recommended otherwise as the HdrHistogram strains systems like Prometheus with excessive numbers of histogram buckets. Note that HdrHistograms are slated for full deprecation in upcoming releases. [#96514][#96514]
```
[the remark this answers is not in the document]
Comment (rmloveland) [reply _includes/releases/v22.2/v22.2.4.md:27]: Updated, thank you!

File: _includes/releases/v22.2/v22.2.4.md [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+
+<h3 id="v22-2-4-db-console-changes">DB Console changes</h3>
+
+- Remove `reset sql stats` and `reset index stats` from the [SQL Activity Page](../v22.2/ui-transactions-page.html) when the user is not an [`admin` user](../v22.2/security-reference/authorization.html#admin-role). [#95331][#95331]
+- The [statement fingerprint details](../v22.2/ui-statements-page.html#sql-statement-fingerprints) page in the DB Console no longer get stuck infinitely loading after 5 minutes. [#95246][#95246]
```
[the remark this answers is not in the document]
Comment (rmloveland) [reply _includes/releases/v22.2/v22.2.4.md:32]: Fixed!

File: _includes/releases/v22.2/v22.2.4.md [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+- Graphs in the [DB Console](../v22.2/ui-overview.html) can be clicked on to toggle legend "stickiness" and make the points stop following the mouse. This makes it easier to read dense graphs with many series plotted together. [#94787][#94787]
+- Remove `reset sql stats` from the [Transactions page](../v22.2/ui-transactions-page.html) for non-admin users. [#95520][#95520]
```
[the remark this answers is not in the document]
Comment (rmloveland) [reply _includes/releases/v22.2/v22.2.4.md:34]: Updated!

File: _includes/releases/v22.2/v22.2.4.md [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+- Hide the apply option for [index recommendation](../v22.2/ui-databases-page.html#index-recommendations) when user is not an [`admin` user](../v22.2/security-reference/authorization.html#admin-role). [#95598][#95598]
```
[the remark this answers is not in the document]
Comment (rmloveland) [reply _includes/releases/v22.2/v22.2.4.md:35]: Fixed!

File: _includes/releases/v22.2/v22.2.4.md [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+- Reduced contention between registering and deregistering [sessions](../v22.2/show-sessions.html), as well as [query cancellation](../v22.2/cancel-query.html) inside a session. [#95626][#95626]
+- Fixed the `pg_get_indexdef` function so that it shows the expression used to define an [expression-based index](../v22.2/expression-indexes.html). Also fixed a bug where the function was previously including columns stored by the index, which was incorrect. [#95584][#95584]
+- Since v22.2, [`DEFAULT` expressions](../v22.2/default-value.html) can have a type which differs from the type of the [column](../v22.2/show-columns.html) so long as the expression's type can be cast in an assignment context. Unfortunately, this code had a bug when adding new columns. The code in the backfill logic was not sophisticated enough to know to add the cast; when such a default expression was added to a new column it would result in a panic during the backfill. This bug has now been fixed. [#95451][#95451]
```
[the remark this answers is not in the document]
Comment (rmloveland) [reply _includes/releases/v22.2/v22.2.4.md:46]: Fixed, nice edit, thanks!

File: _includes/releases/v22.2/v22.2.4.md (hunk 1 above)
Comment (rmloveland) [reply _includes/releases/v22.2/v22.2.4.md:49]: Updated

File: _includes/releases/v22.2/v22.2.4.md (hunk 2 above)
Comment (rmloveland) [reply _includes/releases/v22.2/v22.2.4.md:53]: Fixed

File: _includes/releases/v22.2/v22.2.4.md [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+- Prometheus histograms will now export more buckets across the board to improve precision & fidelity of information reported by histogram metrics, such as quantiles, as shown in multiple pages in the [DB Console](../v22.2/ui-overview.html). This will lead to an increase in storage requirements to process these histogram metrics in downstream systems like Prometheus, but should still be a marked improvement when compared to the legacy `HdrHistogram` model. If users have issues with the precision of these bucket boundaries, they can set the environment variable `COCKROACH_ENABLE_HDR_HISTOGRAMS=true` to revert to using the legacy `HdrHistogram` model instead, although this is not recommended otherwise as the HdrHistogram strains systems like Prometheus with excessive numbers of histogram buckets. Note that HdrHistograms are slated for full deprecation in upcoming releases. [#96514][#96514]
+
+<h3 id="v22-2-4-db-console-changes">DB Console changes</h3>
+
+- Remove `reset sql stats` and `reset index stats` from the [SQL Activity Page](../v22.2/ui-transactions-page.html) when the user is not an [`admin` user](../v22.2/security-reference/authorization.html#admin-role). [#95331][#95331]
```
[the remark this answers is not in the document]
Comment (rmloveland) [reply _includes/releases/v22.2/v22.2.4.md]: Fixed

**Comment by rmloveland** [comment]
Thank you for the review!

**Pull request merged**